### PR TITLE
Add changes to RequestDataSupplier

### PR DESCRIPTION
**Pull request by dagnir**
*Issue #, if available:*

*Description of changes:*
 - Remove unneeded extension of Function; replace with
   @FunctionalInterface annotation

 - Change getRequestBytes to accept ByteBuffer for more flexibility in
   how the callee supplies data (no need to fill buffer entirely)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinboswell** [comment]
Please push this as a branch off aws-crt-java, instead of a fork so CI can run :)

**Review by justinboswell — COMMENTED** [review]
Java code LGTM, let's just get this through CI.

File: s3-native-client/src/main/java/com/amazonaws/s3/RequestDataSupplier.java [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
+@FunctionalInterface
+public interface RequestDataSupplier extends OperationHandler{
     /**
-     * Gives the supplier implement an array that must be fully populated with sequences of bytes, representing
-     * the data to be sent with the operation's request
+     * Called to retrieve data from this supplier. The supplier must write the
+     * available bytes to the given {@code ByteBuffer}.
      * 
-     * Return true if all of the request data has been supplied
+     * Return true if all of the request data has been supplied,
```
Comment (rccarper) [inline s3-native-client/src/main/java/com/amazonaws/s3/RequestDataSupplier.java:11]: Looks like the comma is by accident? :)

**Review by rccarper — APPROVED** [review]

**Comment by dagnir** [comment]
Closing in favor of #298 which doesn't have my name in the branch name.

File: s3-native-client/src/main/java/com/amazonaws/s3/RequestDataSupplier.java (hunk 1 above)
Comment (dagnir) [reply s3-native-client/src/main/java/com/amazonaws/s3/RequestDataSupplier.java:11]: Fixed in #298

**Pull request closed without merging**